File: imu_calib_python/imu_calib/calib/analyze_gyro_allan.py
from __future__ import annotations

import numpy as np

from .analyze_allan_common import analyze_allan_common


def analyze_gyro_allan(t: np.ndarray, gyro: np.ndarray, **kwargs) -> dict:
    """Compute a basic Allan deviation estimate for gyroscope data."""
    return analyze_allan_common(t, gyro, "gyro", **kwargs)

File: imu_calib_matlab/src/analysis/analyze_gyro_allan.m
function allan = analyze_gyro_allan(t, gyro, varargin)
%ANALYZE_GYRO_ALLAN Compute a basic Allan deviation estimate for gyro data.
% Usage:
%   allan = analyze_gyro_allan(t, gyro)

allan = analyze_allan_common(t, gyro, 'gyro', varargin{:});
end
